File: functional-tests/gateway-ft/src/test/java/com/generoso/ft/gateway/client/model/Endpoint.java
package com.generoso.ft.gateway.client.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor(access = AccessLevel.PACKAGE)
public enum Endpoint {

    PRIVATE_INFO("/private/info", "GET"),
    PRIVATE_HEALTH_CHECK("/private/health", "GET"),
    PRIVATE_METRICS("/private/metrics", "GET"),
    GET_APPS("/eureka/apps", "GET"),
    ADD_APP("/eureka/apps", "POST");

    private final String path;
    private final String method;;
}

File: sales-catalog-ft/src/test/kotlin/com/generoso/ft/salescatalog/client/model/Endpoint.kt
package com.generoso.ft.salescatalog.client.model

enum class Endpoint(val path: String, val method: String) {
    PRIVATE_INFO("/sales-catalog/private/info", "GET"),
    PRIVATE_HEALTH_CHECK("/sales-catalog/private/health", "GET"),
    PRIVATE_METRICS("/sales-catalog/private/metrics", "GET"),
    PRIVATE_OPEN_API("/sales-catalog/private/api-docs", "GET"),
    PRIVATE_SWAGGER_REDIRECTION("/sales-catalog/private/swagger", "GET"),
    PRIVATE_SWAGGER_UI("/sales-catalog/private/swagger-ui/index.html", "GET"),
    PRODUCT_POST("/sales-catalog/v1/products", "POST"),
    PRODUCT_GET("/sales-catalog/v1/products", "GET")
}